File: app/src/main/java/com/upfinder/voicetodo/task/EventsDialogActivity.java
package com.upfinder.voicetodo.task;


import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import com.upfinder.voicetodo.R;

import org.json.JSONArray;
import org.json.JSONObject;


public class EventsDialogActivity extends Activity {

    private TextView mEventTitle;
    private Button mPreBtn;
    private Button mNextBtn;
    private RadioGroup mRgroup;
    private RadioButton mRbSuc;
    private RadioButton mRbFail;
    private RadioButton mRbIgnore;
    private JSONArray mEventsArrary;
    private JSONObject mCurrentEvent;
    private int mIndex = 0;
    private boolean isFinish = false;
    private View mViewOutSide;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewOutSide = LayoutInflater.from(this).inflate(R.layout.events_dialog,null);
        setContentView(mViewOutSide);
        String events = getIntent().getStringExtra("events");
        String taskid = getIntent().getStringExtra("taskId");
        setEvents(events);
        initView();

    }

    public void setEvents(String events){
        try {
            mEventsArrary = new JSONArray(events);
            mIndex = 0;
            refreshUI();
        }catch (Exception e){
            e.printStackTrace();
        }

    }

    public void initView() {

        mEventTitle = findViewById(R.id.eventTitle);
        mRgroup = findViewById(R.id.rgEventState);
        mPreBtn = findViewById(R.id.btPreEvent);
        mNextBtn = findViewById(R.id.btNextEvent);
        mRbSuc = findViewById(R.id.rbSuccess);
        mRbFail = findViewById(R.id.rbFail);
        mRbIgnore = findViewById(R.id.rbIgnore);

        mRgroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup radioGroup, int i) {
                try{
                    switch (i){
                        case R.id.rbSuccess:
                            mCurrentEvent.put("state",AddTaskActivity.EVENTS_STATE_SUCCESS);
                            break;
                        case R.id.rbFail:
                            mCurrentEvent.put("state",AddTaskActivity.EVENTS_STATE_FAIL);
                            break;
                        case R.id.rbIgnore:
                            mCurrentEvent.put("state",AddTaskActivity.EVENTS_STATE_IGNORE);
                            break;
                    }
                    mEventsArrary.put(mIndex,mCurrentEvent);
                }catch (Exception e){
                    e.printStackTrace();
                }
            }
        });

        mPreBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (isFinish)
                {
                    saveEvent();
                }else{
                    mIndex --;
                    refreshUI();
                }
            }
        });

        mNextBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (isFinish)
                {
                    clearEvent();
                }else{
                    mIndex ++;
                    refreshUI();
                }
            }
        });
    }

    //刷新界面
    public void refreshUI(){
        mHandler.removeCallbacksAndMessages(null);
        mHandler.sendEmptyMessage(HANDLER_MSG_REFRESHUI);
    }

    //保存事件
    public void saveEvent(){
        finish();
    };

    //清除事件
    public void clearEvent(){
        finish();
    }

    public static final int HANDLER_MSG_REFRESHUI = 1;

    public Handler mHandler = new Handler(new Handler.Callback() {
        @Override
        public boolean handleMessage(Message msg) {
            switch (msg.what){
                case  HANDLER_MSG_REFRESHUI:
                    try {
                        mCurrentEvent = mEventsArrary.getJSONObject(mIndex);
                        String event = mCurrentEvent.getString("event");
                        int state = mCurrentEvent.getInt("state");
                        mEventTitle.setText(event);
                        switch (state){
                            case AddTaskActivity.EVENTS_STATE_DEFAULT:
                            case AddTaskActivity.EVENTS_STATE_SUCCESS:
                                mRbSuc.setChecked(true);
                                mRbIgnore.setChecked(false);
                                mRbFail.setChecked(false);
                                break;
                            case AddTaskActivity.EVENTS_STATE_FAIL:
                                mRbSuc.setChecked(false);
                                mRbIgnore.setChecked(false);
                                mRbFail.setChecked(true);
                                break;
                            case AddTaskActivity.EVENTS_STATE_IGNORE:
                                mRbSuc.setChecked(false);
                                mRbIgnore.setChecked(true);
                                mRbFail.setChecked(false);
                                break;
                        }
                        //长度相等时，显示“保存”和“清除”
                        if ((mIndex+1) == mEventsArrary.length())
                        {
                            isFinish = true;
                            mPreBtn.setText("保存");
                            mNextBtn.setText("清除");
                        }else if(mIndex == 0)
                        {
                            isFinish = false;
                            mPreBtn.setEnabled(false);
                            mPreBtn.setText("上一项");
                            mNextBtn.setText("下一项");
                        }else {
                            isFinish = false;
                            mPreBtn.setEnabled(true);
                            mNextBtn.setEnabled(true);
                            mPreBtn.setText("上一项");
                            mNextBtn.setText("下一项");
                        }
                    }catch (Exception e){
                        e.printStackTrace();
                    }
                    break;
            }
            return false;
        }
    });
}

File: app/src/main/java/com/upfinder/voicetodo/task/EventsDialogActivity.kt
package com.upfinder.voicetodo.task

import android.app.Activity
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.TextView
import com.upfinder.voicetodo.MyApplication
import com.upfinder.voicetodo.R
import com.upfinder.voicetodo.worker.BaiduTTsApi
import kotlinx.android.synthetic.main.events_dialog.*
import org.json.JSONArray
import org.json.JSONObject

class EventsDialogActivity : Activity(){

    private var mEventsArrary: JSONArray? = null
    private var mCurrentEvent: JSONObject? = null
    private var mIndex = 0
    private var isFinish = false
    private var mTaskId : String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.events_dialog)
        val events = intent.getStringExtra("events")
        mTaskId = intent.getStringExtra("taskId")
        setEvents(events)
        initView()
    }

    fun setEvents(events: String) {
        try {
            mEventsArrary = JSONArray(events)
            mIndex = 0
            refreshUI()
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    fun initView() {

        rgEventState.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { radioGroup, i ->
            try {
                when (i) {
                    R.id.rbSuccess -> mCurrentEvent!!.put("state", AddTaskActivity.EVENTS_STATE_SUCCESS)
                    R.id.rbFail -> mCurrentEvent!!.put("state", AddTaskActivity.EVENTS_STATE_FAIL)
                    R.id.rbIgnore -> mCurrentEvent!!.put("state", AddTaskActivity.EVENTS_STATE_IGNORE)
                }
                mEventsArrary!!.put(mIndex, mCurrentEvent)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        })

        btPreEvent.setOnClickListener(View.OnClickListener {
            if (isFinish) {
                saveEvent()
            } else {
                mIndex--
                refreshUI()
            }
        })

        btNextEvent.setOnClickListener(View.OnClickListener {
            if (isFinish) {
                clearEvent()
            } else {
                mIndex++
                refreshUI()
            }
        })

        eventCancel.setOnClickListener(View.OnClickListener {
            finish();
        })
    }

    //刷新界面
    fun refreshUI() {
        mHandler.removeCallbacksAndMessages(null)
        mHandler.sendEmptyMessage(HANDLER_MSG_REFRESHUI)
    }

    //保存事件
    fun saveEvent() {
        finish()
    };

    //清除事件
    fun clearEvent() {
        MyApplication.getTasksLocalDataSourceInstance().deleteTask(mTaskId)
        finish()
    }

    val HANDLER_MSG_REFRESHUI = 1

    var mHandler = Handler(Handler.Callback { msg ->
        when (msg.what) {
            HANDLER_MSG_REFRESHUI -> try {
                mCurrentEvent = mEventsArrary!!.getJSONObject(mIndex)
                val event = mCurrentEvent!!.getString("event")
                val state = mCurrentEvent!!.getInt("state")
                eventTitle.setText(event)
                when (state) {
                    AddTaskActivity.EVENTS_STATE_DEFAULT, AddTaskActivity.EVENTS_STATE_SUCCESS -> {
                        rbSuccess.setChecked(true)
                        rbIgnore.setChecked(false)
                        rbFail.setChecked(false)
                    }
                    AddTaskActivity.EVENTS_STATE_FAIL -> {
                        rbSuccess.setChecked(false)
                        rbIgnore.setChecked(false)
                        rbFail.setChecked(true)
                    }
                    AddTaskActivity.EVENTS_STATE_IGNORE -> {
                        rbSuccess.setChecked(false)
                        rbIgnore.setChecked(true)
                        rbFail.setChecked(false)
                    }
                }
                MyApplication.getInstance()
                //长度相等时，显示“保存”和“清除”
                if (mIndex + 1 == mEventsArrary!!.length()) {
                    isFinish = true
                    btPreEvent.setEnabled(true)
                    btPreEvent.setText("保存")
                    btNextEvent.setText("清除")
                } else if (mIndex == 0) {
                    isFinish = false
                    btPreEvent.setEnabled(false)
                    btPreEvent.setText("上一项")
                    btNextEvent.setText("下一项")
                } else {
                    isFinish = false
                    btPreEvent.setEnabled(true)
                    btNextEvent.setEnabled(true)
                    btPreEvent.setText("上一项")
                    btNextEvent.setText("下一项")
                }
                MyApplication.getTTs().playTTS(event)
            } catch (e: Exception) {
                e.printStackTrace()
            }

        }
        false
    })
}